src/cycles/incremental.jl

```diff
@@ -0,0 +1,227 @@
+using Base.Iterators: repeated
+
+# Abstract Interface
+
+"""
+    abstract type IncrementalCycleTracker
+
+The supertype for incremental cycle detection problems. The abstract type
+constructor IncrementalCycleTracker(G) may be used to automatically select
+a specific incremental cycle detection algorithm. See [`add_edge_checked!`](@ref)
+for a usage example.
+"""
+abstract type IncrementalCycleTracker{I} <: AbstractGraph{I} end
+
+function (::Type{IncrementalCycleTracker})(s::AbstractGraph{I}; in_out_reverse=nothing) where {I}
```

Comment:
Why not `in_out_reverse=false`?

Reply:
The idea is to be able to detect somebody explicitly passing in `in_out_reverse=false`, in case we want to change these options later, since it's a bit ad hoc. It helps to have a value that represent "no user preference" such that people writing wrappers can use that as a default value if they want without incorrectly signaling user intent. E.g. if we wanted to deprecate this argument, we'd check `isa(in_out_reverse, Bool)`.